### PR TITLE
chore: add list testing

### PR DESCRIPTION
**Pull request by pgautier404**
More testing is on the way for all collection data types regarding struct defaults, nil requests, and related issues. This commit brings the list type testing to parity with the other types to this point.

### CODE REVIEW

File: momento/list_test.go [hunk 1]
```diff
@@ -0,0 +1,537 @@
...
+	values, expected := getValueAndExpectedValueLists(numItems)
+	for _, value := range values {
+		Expect(
+			sharedContext.Client.ListPushBack(sharedContext.Ctx, &ListPushBackRequest{
+				CacheName: sharedContext.CacheName,
+				ListName:  sharedContext.CollectionName,
+				Value:     value,
+			}),
+		).To(BeAssignableToTypeOf(&ListPushBackSuccess{}))
+	}
```
Comment (schwern) [inline momento/list_test.go:40]: Can this be replaced with ListConcatenateBack?

File: momento/list_test.go [hunk 2]
```diff
@@ -0,0 +1,537 @@
...
+
+	BeforeEach(func() {
+		sharedContext = NewSharedContext()
+		sharedContext.CreateDefaultCache()
+		DeferCleanup(func() {
+			sharedContext.Close()
+		})
+	})
+
+	DescribeTable("try using invalid cache and list names",
```
Comment (schwern) [inline momento/list_test.go:55]: This is missing most of the list methods.

Comment (pgautier404) [reply momento/list_test.go:55]: Good catch, thanks.

File: momento/list_test.go (hunk 1 above)
Comment (pgautier404) [reply momento/list_test.go:40]: Why yes, yes it can.

**Review by schwern — APPROVED** [review]
